File: autotest/Tests/src/test/scala/equellatests/domain/ItemId.scala
package equellatests.domain

import java.util.UUID

import io.circe.generic.semiauto._

case class ItemId(uuid: UUID, version: Int)

object ItemId {

  def random = ItemId(UUID.randomUUID(), 1)

  implicit def fromJava(oldId: com.tle.webtests.pageobject.viewitem.ItemId): ItemId =
    ItemId(UUID.fromString(oldId.getUuid), oldId.getVersion)

  implicit val itemIdEnc = deriveEncoder[ItemId]
  implicit val itemIdDec = deriveDecoder[ItemId]

}

File: Source/Tools/Cacher/src/com/dytech/edge/cache/ItemId.java
package com.dytech.edge.cache;

public class ItemId {
  private String uuid;
  private int version;

  public ItemId(String uuid, int version) {
    this.uuid = uuid;
    this.version = version;
  }

  public String getUuid() {
    return uuid;
  }

  public void setUuid(String uuid) {
    this.uuid = uuid;
  }

  public int getVersion() {
    return version;
  }

  public void setVersion(int version) {
    this.version = version;
  }

  @Override
  public String toString() {
    return uuid + '/' + version;
  }
}
